app/services/api/v3/actor_node/exporting_companies_plot.rb

```diff
@@ -0,0 +1,117 @@
+module Api
+  module V3
+    module ActorNode
+      class ExportingCompaniesPlot
+        include Api::V3::Profiles::AttributesInitializer
+
+        def initialize(context, year, node)
+          @context = context
+          @year = year
+          @node = node
+          @volume_attribute = Dictionary::Quant.instance.get('Volume')
+          raise 'Quant Volume not found' unless @volume_attribute.present?
+          initialize_attributes(attributes_list)
+        end
+
+        def call
+          unit = @volume_attribute.unit
+          value_divisor = 1
+          if unit.casecmp('tn').zero?
+            unit = 'Kiloton'
+            value_divisor = 1000
+          end
+
+          stats = Api::V3::Profiles::FlowStatsForNodeType.new(
+            @context, @year, @node.node_type.name
+          )
+          production_totals = stats.nodes_with_flows_totals(@volume_attribute)
+          attribute_totals = stats.nodes_with_flows_totals_for_attributes(
+            @attributes.map { |attribute_hash| attribute_hash[:attribute] }
+          )
+
+          attribute_indexes = Hash[
+            @attributes.map.each_with_index do |attribute_hash, idx|
+              [attribute_hash[:attribute_name], idx]
+            end
+          ]
+
+          attribute_totals_hash = {}
+          production_totals.each do |total|
+            attribute_totals_hash[total['node_id']] ||= Array.new(
+              @attributes.size
+            )
+          end
+          attribute_totals.each do |total|
+            attribute_idx = attribute_indexes[total['attribute_name']]
+            if attribute_totals_hash.key?(total['node_id'])
+              attribute_totals_hash[total['node_id']][attribute_idx] = total['value']
+            end
+          end
+
+          exports = production_totals.map do |total|
+            {
+              name: total['name'],
+              id: total['node_id'],
+              y: total['value'].to_f / value_divisor,
+              x: attribute_totals_hash[total['node_id']]
+            }
+          end
+
+          {
+            companies_sourcing: {
+              dimension_y: {
+                name: 'Trade Volume', unit: unit
+              },
+              dimensions_x: @attributes.map do |attribute_hash|
+                attribute_hash.slice(:name, :unit)
+              end,
+              companies: exports
+            }
+          }
+        end
+
+        private
+
+        def attributes_list
```

Comment:
We should pass this info on to Clement, so he knows these attributes are required.
Also, can't we load names and units from the DB? this info should be there already.

Reply:
This info is not in the db. We have designed tables where these lists of attributes could be kept, and I guess it would be possible to store them there, but without a tool to update them this has little value. We've discussed not implementing the fully dynamic profile pages with configurable attributes in this phase, so I kept it as is. Good point about providing Clement with a full list of currently used attributes.